lib/storage/src/content_manager/toc/collection_meta_ops.rs

```diff
@@ -364,7 +372,12 @@ impl TableOfContent {
                     },
                 )
                 .await?;
-                self.handle_transfer(collection_id, ShardTransferOperations::Start(transfer))
+
+                let mut new_transfer = transfer.clone();
+                // Preserve sync flag from the old transfer
+                new_transfer.sync = old_transfer.sync;
+
```

Comment:
This is new stuff from my side, might require extra look

Reply:
Shall we do an OR, to sync if the old OR the new one has sync=true?

```suggestion
                // Preserve sync flag from the old transfer
                let mut new_transfer = transfer.clone();
                new_transfer.sync |= old_transfer.sync;

```

Reply:
I've accidentally hit the green button.

I'm handling this here: https://github.com/qdrant/qdrant/pull/3728